src/DocumentDB.ChangeFeedProcessor.IntegrationTests/IntegrationTest.cs

```diff
@@ -93,82 +66,58 @@ public class IntegrationTestCollection : ICollectionFixture<IntegrationTestFixtu
     /// </remarks>
     [Trait("Category", "Integration")]
     [Collection("Integration tests")]
-    public class IntegrationTest: IDisposable
+    public class IntegrationTest: IAsyncLifetime
     {
-        private const string leaseCollectionInfoPropertyName = "leaseCollectionInfo";
         protected static int monitoredOfferThroughput;
         protected static int leaseOfferThroughput;
         protected static readonly TimeSpan changeWaitTimeout = TimeSpan.FromSeconds(30);
 
-        /// <summary>
-        /// This dictionary has one entry per derived class.
-        /// </summary>
-        private static object testClassesSyncRoot = new object();
-
         IntegrationTestFixture fixture;
 
         protected DocumentCollectionInfo LeaseCollectionInfo
         {
-            get { return (DocumentCollectionInfo)this.fixture.Properties[leaseCollectionInfoPropertyName]; }
-            set
-            {
-                lock (this.ClassData.testContextSyncRoot)
-                {
-                    this.fixture.Properties[leaseCollectionInfoPropertyName] = value;
-                }
-            }
+            get;
+            private set;
         }
 
-        public TestClassData ClassData
+        protected DocumentCollectionInfo MonitoredCollectionInfo
         {
-            get { return this.fixture.testClasses[this.GetType().Name]; }
+            get;
+            private set;
         }
-        
+
+        protected readonly bool IsPartitionedCollection;
+
+        protected readonly bool IsPartitionedLeaseCollection;
+
         public IntegrationTest(
-            IntegrationTestFixture fixture,
-            Type testClassType,
             bool isPartitionedCollection = true,
             bool isPartitionedLeaseCollection = false)
         {
-            this.fixture = fixture;
-            if (!this.fixture.testClasses.ContainsKey(testClassType.Name))
-            {
-                this.fixture.testClasses[testClassType.Name] = new TestClassData(
-                    GetTestCount(testClassType),
```

Comment:
Is GetTestCount method needed now? Please remove if not.

Reply:
Removed